moped-editor/src/utils/mapHelpers.js

```diff
@@ -1056,3 +1055,38 @@ export const queryCtnFeatureService = async function(projectExtentId) {
       return null;
     });
 };
+
+/**
+ * Immitate a "drawn point" feature from a traffic signal goejosn feature. Sets required
+ * fields so that featureCollection can be used in the DB mutation on submit
+ * @param {Object} signal - A GeoJSON feature or a falsey object (e.g. "" from empty input)
+ * @return {Object} A geojson feature collection with the signal feature or 0 features
+ */
+export const signalToFeatureCollection = signal => {
+  let featureCollection = {
+    type: "FeatureCollection",
+    features: [],
+  };
+  if (signal) {
+    /* 
+    / preserves the feature's previous UUID if it's being edited. we are **not** preserving
+    / any other feature properties when the feature is edited. so, for example, if the user
+    / edits a signal component and the signal geometry in socrata has since changed, the new
+    / geometry will be saved.
+    */
+    const featureUUID = signal.id || uuidv4();
```

Comment:
No action required on here, but I wonder if `signal.id` could potentially throw a WSOD in some unlikely scenario where (signal) is a value and not an object, or it is an object but does not have the "id" key. I may be mistaken, but I wonder if the use of conditional chaining is warranted in this case.

Reply:
you're right. should never happen, but it's risky to rely on perfect data from socrata 100% of the time. i'll follow your suggestion.